### PR TITLE
eth/protocols/eth,p2p/discover: remove unnecessary checks

### PR DESCRIPTION
**Pull request by undefinedor**

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -793,7 +793,7 @@ func configBlockEqual(x, y *big.Int) bool {
 		return y == nil
 	}
 	if y == nil {
-		return x == nil
+		return false
```
Comment (holiman) [inline params/config.go:796]: current code is more robust, new code depends on order. Neither is IMO more readable than the other. please revert

File: params/config.go [hunk 2]
```diff
@@ -819,7 +819,7 @@ func configTimestampEqual(x, y *uint64) bool {
...
```
Comment (holiman) [inline params/config.go:822]: current code is more robust, new code depends on order. Neither is IMO more readable than the other. please revert

**Review by holiman — CHANGES_REQUESTED** [review]

**Comment by undefinedor** [comment]
@holiman I have already reverted.

File: params/config.go (hunk 1 above)
Comment (undefinedor) [reply params/config.go]: done

File: params/config.go (hunk 2 above)
Comment (undefinedor) [reply params/config.go]: done

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**